pointcloud/pointcloud_file.go

```diff
@@ -538,69 +574,117 @@ func readPCDHelper(inRaw io.Reader, pctype PCType) (PointCloud, error) {
 	}
 }
 
-func readPCDAscii(in *bufio.Reader, header pcdHeader, pc PointCloud) (PointCloud, error) {
-	for i := 0; i < int(header.points); i++ {
-		line, err := in.ReadString('\n')
+func extractPCDPointASCII(in *bufio.Reader, header pcdHeader, i int) (PointAndData, error) {
+	line, err := in.ReadString('\n')
+	if err != nil {
+		return PointAndData{}, err
+	}
+	line = strings.TrimSpace(line)
+	tokens := strings.Split(line, " ")
+	if len(tokens) != int(header.fields) {
+		return PointAndData{}, fmt.Errorf("unexpected number of fields in point %d", i)
+	}
+	point := make([]float64, len(tokens))
+	for j, token := range tokens {
```

Comment:
[nit] maybe consider changing the variables to i given that j is usually used within a second nested loop, which is not the case for this function?

Reply:
See previous comment